File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution(object):
    def cnt(self,n):
        t=n
        s=0
        while t>0:
            s+=(t%10)
            t=t//10
        return s
    def smallestIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for i in range(len(nums)):
            t=self.cnt(nums[i])
            if(t==i):
                return i
        return -1

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.cpp
class Solution {
public:
    int ds(int n)
    {
        int s=0;
        while(n>0)
            {
                int d=n%10;
                s+=d;
                n=n/10;
            }
        return s;
    }
    int smallestIndex(vector<int>& nums) {
        for(int i=0;i<nums.size();i++)
            {
                int dd=ds(nums[i]);
                if(dd==i)
                {
                    return i;
                }
            }
        return -1;
        
    }
};